op-node/p2p/gossip.go

```diff
@@ -441,3 +443,11 @@ func LogTopicEvents(ctx context.Context, log log.Logger, evHandler *pubsub.Topic
 		}
 	}
 }
+
+type gossipTracer struct {
+	m *metrics.Metrics
+}
+
+func (g *gossipTracer) Trace(evt *pb.TraceEvent) {
+	g.m.RecordGossipEvent(int32(*evt.Type))
+}
```

Comment:
```suggestion
func (g *gossipTracer) Trace(evt *pb.TraceEvent) {
    if g.m != nil {
	    g.m.RecordGossipEvent(int32(*evt.Type))
	}
}
```
We could make the metrics optional like this.

Reply:
And maybe not in this PR, but based on the event type we can decide to collect other metrics. E.g. if a peer subscribes to a topic, we can track the number of peers on that topic. Or e.g. update a gauge with number of peers in our mesh.